service/routers/users.js

```diff
@@ -0,0 +1,50 @@
+import { Router } from 'express'
+import { User } from '../models/User.js'
+
+const usersRouter = Router()
+
+const getUsers = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser") {
+    const users = await User.find({})
+    res.send(users)
+  } else {
+    res.status(404).send('only admin and superuser can access this route')
```

Comment:
This is the wrong status code to send. 404 means 'not found'. There is a status code for 'Forbidden'. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses
Review ALL of your status code returns for limited access.